dev/TestMailer.php

```diff
@@ -4,14 +4,14 @@
  * @subpackage email
  */
 class TestMailer extends Mailer {
-	protected $emailsSent = array();
+	protected static $emailsSent = array();
```

Comment:
I think that's what `clearEmails` is for (flushing emails, that is)? Since TestMailer isn't enforced to be a singleton during a test, you can get multiple instances of it, where one will have Emails stores and another one doesn't. This was the case for me when I triggered an Email via `FunctionalTest::post` (see my `testForgotPasswordEmaling` test). There seem to be two different instances of TestMailer being instantiated in that case and the Mailer used within the tests itself won't contain any mails, unless `$emailsSent` is a static member.

I think overall a static member makes sense, since you can't enforce the mailer instance via `Mailer::set_mailer`. Tests that send Emails should use `$this->clearEmails` at the start… I don't see any issues with this, unless you're starting to run tests in parallel (and even then, you can find your sent Email using the existing API).